examples/implementation_languages/coffeescript/coffeescript_json.coffee

```diff
@@ -0,0 +1,97 @@
+chevrotain = require 'chevrotain'
+Token = chevrotain.Token
+Lexer = chevrotain.Lexer
+Parser = chevrotain.Parser
+
+class True extends Token
+  @PATTERN: /true/
+
+class False extends Token
+  @PATTERN: /false/
+
+class Null extends Token
+  @PATTERN: /null/
+
+class LCurly extends Token
+  @PATTERN: /{/
+
+class RCurly extends Token
+  @PATTERN: /}/
+
+class LSquare extends Token
+  @PATTERN: /\[/
+
+class RSquare extends Token
+  @PATTERN: /]/
+
+class Comma extends Token
+  @PATTERN: /,/
+
+class Colon extends Token
+  @PATTERN: /:/
+
+class StringLiteral extends Token
+  @PATTERN: /"(:?[^\\"]+|\\(:?[bfnrtv"\\/]|u[0-9a-fA-F]{4}))*"/
+
+class NumberLiteral extends Token
+  @PATTERN: /-?(0|[1-9]\d*)(\.\d+)?([eE][+-]?\d+)?/
+
+class WhiteSpace extends Token
+  @PATTERN = /\s+/
+  @GROUP = Lexer.SKIPPED
+
+allTokens = [WhiteSpace, NumberLiteral, StringLiteral, LCurly, RCurly, LSquare, RSquare, Comma, Colon, True, False, Null]
+JsonLexer = new Lexer allTokens
+
+class JsonParserCoffeeScript extends Parser
+  constructor: (input) ->
+    super input, allTokens
+
+    @RULE "json", =>
+      @.OR [
+        {ALT: => @.SUBRULE @.object}
+        {ALT: => @.SUBRULE @.array}
+      ]
+
+    @RULE "object", =>
+      @.CONSUME LCurly
+      @.MANY_SEP Comma, =>
+        @.SUBRULE2 @.objectItem
+      @.CONSUME RCurly
+
+    @RULE "objectItem", =>
+      @.CONSUME StringLiteral
```

Comment:
That's helpful. Thank you for taking the time to write examples of multiple syntax flavors. If only we all spent the time doing that :)

I've begun writing my README's with JavaScript (ES5) examples and then an alternate CoffeeScript README. I want to do ES6 versions, but, when I try I find my ES6 skills are enough to read it, but, writing it, not so much.

Anyway, one note for CoffeeScript:

```coffeescript
#  instead of this:
chevrotain = require 'chevrotain'
Token = chevrotain.Token
Lexer = chevrotain.Lexer
Parser = chevrotain.Parser

# we can use destructuring like this:
{Token, Lexer, Parser} = require 'chevrotain'
```

Also, I see the revised version of the `RULE()` function. That's nice. Thank you.